eZ/Publish/Core/MVC/Symfony/Controller/Content/QueryController.php

```diff
@@ -81,4 +88,67 @@ private function runQuery(ContentView $view, $method)
         );
         $view->addParameters([$view->getParameter('query')['assign_results_to'] => $searchResults]);
     }
+
+    /**
+     * @param \eZ\Publish\Core\MVC\Symfony\View\ContentView $view
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @return \eZ\Publish\Core\MVC\Symfony\View\ContentView
+     */
+    public function pagingQueryAction(ContentView $view, Request $request)
+    {
+        $this->runPagingQuery($view, $request);
+
+        return $view;
+    }
+
+    /**
+     * @param \eZ\Publish\Core\MVC\Symfony\View\ContentView $view
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     */
+    private function runPagingQuery(ContentView $view, Request $request)
+    {
+        $queryParameters = $view->getParameter('query');
+
+        $limit = $this->getParameter('limit', 10, $queryParameters);
+        $page_param = $this->getParameter('page_param', 'page', $queryParameters);
+
+        $page = $request->get($page_param, 1);
+
+        $pager = new Pagerfanta(
+            $this->getAdapter($this->contentViewQueryTypeMapper->map($view))
+        );
+
+        $pager->setMaxPerPage($limit);
+        $pager->setCurrentPage($page);
+
+        $view->addParameters([$queryParameters['assign_results_to'] => $pager]);
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Query $query
+     * @return \Pagerfanta\Adapter\AdapterInterface
+     */
+    private function getAdapter(Query $query): AdapterInterface
+    {
+        if ($query instanceof LocationQuery) {
+            return new LocationSearchHitAdapter($query, $this->searchService);
+        } else {
+            return new ContentSearchHitAdapter($query, $this->searchService);
+        }
+    }
+
+    /**
+     * @param $name
+     * @param $defaultValue
+     * @param $parameters
+     * @return mixed
+     */
+    private function getParameter($name, $defaultValue, $parameters)
```

Comment:
Missing return type after `:`. I see it might be difficult, hence this method is not the best choice here :)